README
NAME

    Dist::Zilla::Plugin::ReadmeAnyFromPod - Automatically convert POD to a
    README in any format for Dist::Zilla

VERSION

    version 0.142470

SYNOPSIS

    In your dist.ini

        [ReadmeAnyFromPod]
        ; Default is plaintext README in build dir
    
        ; Using non-default options: POD format with custom filename in
        ; dist root, outside of build. Including this README in version
        ; control makes Github happy.
        [ReadmeAnyFromPod / ReadmePodInRoot]
        type = pod
        filename = README.pod
        location = root
    
        ; Using plugin name autodetection: Produces README.html in root
        [ ReadmeAnyFromPod / HtmlInRoot ]

DESCRIPTION

    Generates a README for your Dist::Zilla powered dist from its
    main_module in any of several formats. The generated README can be
    included in the build or created in the root of your dist for e.g.
    inclusion into version control.

 PLUGIN NAME AUTODETECTION

    If you give the plugin an appropriate name (a string after the slash)
    in your dist.ini, it will can parse the type and location attributes
    from it. The format is "Readme[TYPE]In[LOCATION]". The words "Readme"
    and "In" are optional, and the whole name is case-insensitive. The
    SYNOPSIS section above gives one example.

    When run with location = dist, this plugin runs in the FileMunger phase
    to create the new file. If it runs before another FileMunger plugin
    does, that happens to modify the input pod (like, say, [PodWeaver]),
    the README file contents will be recalculated, along with a warning
    that you should modify your dist.ini by referencing [ReadmeAnyFromPod]
    lower down in the file (the build still works, but is less efficient).

ATTRIBUTES

 type

    The file format for the readme. Supported types are "text", "markdown",
    "pod", and "html". Note that you are not advised to create a .pod file
    in the dist itself, as ExtUtils::MakeMaker will install that, both into
    PERL5LIB and MAN3DIR.

 filename

    The file name of the README file to produce. The default depends on the
    selected format.

 source_filename

    The file from which to extract POD for the content of the README. The
    default is the file of the main module of the dist. If the main module
    has a companion ".pod" file with the same basename, that is used as the
    default instead.

 location

    Where to put the generated README file. Choices are:

    build

      This puts the README in the directory where the dist is currently
      being built, where it will be incorporated into the dist.

    root

      This puts the README in the root directory (the same directory that
      contains dist.ini). The README will not be incorporated into the
      built dist.

    If you want to generate the same README file in both the build
    directory and the root directory, simply generate it in the build
    directory and use the [CopyFilesFromBuild] plugin to copy it to the
    dist root.

 phase

    At what phase to generate the README file. Choices are:

    build

      (Default) This generates the README at 'after build' time. A new
      README will be generated each time you build the dist.

    release

      This generates the README at 'after release' time. Note that this is
      too late to get the file into the generated tarball, and is therefore
      incompatible with location = build. However, this is ideal if you are
      using location = root and only want to update the README upon each
      release of your module.

METHODS

 gather_files

    We create the file early, so other plugins that need to have the full
    list of files are aware of what we will be generating.

 prune_files

    Files with location = root must also be pruned, so that they don't
    sneak into the next build by virtue of already existing in the root
    dir. (The alternative is that the user doesn't add them to the build in
    the first place, with an option to their GatherDir plugin.)

 munge_files

 munge_file

    Edits the content into the requested README file in the dist.

 after_build

    Create the requested README file at build time, if requested.

 after_release

    Create the requested README file at release time, if requested.

 get_readme_content

    Get the content of the README in the desired format.

BUGS AND LIMITATIONS

    Please report any bugs or feature requests to
    rct+perlbug@thompsonclan.org.

SEE ALSO

      * Dist::Zilla::Plugin::ReadmeFromPod - The base for this module

      * Dist::Zilla::Plugin::ReadmeMarkdownFromPod - Functionality subsumed
      by this module

      * Dist::Zilla::Plugin::CopyReadmeFromBuild - Functionality partly
      subsumed by this module

INSTALLATION

    See perlmodinstall for information and options on installing Perl
    modules.

AUTHORS

      * Ryan C. Thompson <rct@thompsonclan.org>

      * Karen Etheridge <ether@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Ryan C. Thompson.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

DISCLAIMER OF WARRANTY

    BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
    FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT
    WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER
    PARTIES PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND,
    EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
    ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH
    YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
    NECESSARY SERVICING, REPAIR, OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
    REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENCE, BE LIABLE
    TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR
    CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
    SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
    RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
    FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
    SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
    DAMAGES.



Changes
Revision history for Dist-Zilla-Plugin-ReadmeAnyFromPod

0.142470  2014-09-04 16:14:01PDT-0700 America/Los_Angeles
    - Use main module pod file by default if available. If your main
      module has its POD in a separate file with the same basename and
      ".pod" extension, then that file will be used automatically as
      the default.

0.142250  2014-08-13 08:34:36PDT-0700 America/Los_Angeles
    - If you pruned away the README(.*) file before it could be populated with
      content, you'd get a fatal error like " Can't call method "name" on an
      undefined value at ...". Now we detect this and throw a more helpful
      (but still fatal) error suggesting you check your PruneFiles
      config. RT #97976. (Dave Rolsky)

0.142180  2014-08-06 09:10:56PDT-0700 America/Los_Angeles
    - add "phase" option, for the ability to generate the README file at
      release time instead of build time (Karen Etheridge)

0.141760  2014-06-25 01:13:06PDT-0700 America/Los_Angeles
    - Load some modules dynamically only when they are used
    - Eliminate implicit dependency on specific version of Path::Class
    - Minor documentation updates

0.141120  2014-04-22 15:35:30 America/Los_Angeles
    - Adjust tests for change in Pod::Markdown output

0.133360  2013-12-02 01:09:18 America/Los_Angeles
    - Require the appropriate version of Path::Tiny

0.133290  2013-11-24 22:12:39 America/Los_Angeles
    - First real release compatible with dzil v5

0.133060  2013-11-02 16:56:50 America/Los_Angeles
    - Add a test for non-utf8 encoded input.

0.133030  2013-10-30 00:53:24 America/Los_Angeles
    - Require Pod::Simple version 3.23 for parse_characters

0.133000  2013-10-27 14:12:47 America/Los_Angeles
    - Fix handling of case where readme contents are regenerated after the
      source .pm is altered after we initially looked at it

0.132970  2013-10-24 12:11:46 America/Los_Angeles
    - Module now works with both Dist::Zilla version 5.X and up (which
      hanldes file encodings) and Dist::Zilla version 4.X and below
      (with no special handling of file encodings at all). This is a
      potential regression for users of Dist::Zilla 4.X, since
      previously this module made some poor attempts at handling
      encoding. If you need correct handling of file encodings, you
      should upgrade Dist::Zilla to version 5.

0.132973  2013-10-23 17:15:29 America/Los_Angeles
    - now using FileGatherer, FileMunger and AfterBuild phases rather than
      InstallerTool

0.132962  2013-10-23 15:45:36 America/Los_Angeles
    - Now compatible *only* with Dist::Zilla version 5 and up
    - Add test for excluding root README files from build

0.131500  2013-05-30 02:03:44 America/Los_Angeles
    - Allow source POD to come from somewhere other than the main
      module (contributed by Hercynium)

0.120120  2012-01-11 22:41:25 PST8PDT
    - Slight improvement to log messages for file overrides
    - Added more tests

0.120053  2012-01-05 13:42:35 PST8PDT
    - Option autodetection now works better. In particular, it now
      works when the plugin is used as part of a bundle.

0.120052  2012-01-05 11:07:05 PST8PDT
    - Added a test to make sure \r characters never appear in any
      output.

0.120051  2012-01-04 23:42:29 PST8PDT
    - Prevent perl from adding \r characters on win32 by writing files
      in binmode. This ensures that the same output is produced on all
      platforms. (contributed by MITHALDU)
    - Prevent modules in corpus directory from being indexed.

0.120040  2012-01-04 11:12:47 PST8PDT
    - Now supports parsing options from given plugin name
    - Added tests, which should all pass

0.103522  2010-12-18 12:44:44 EST5EDT
    - Updated abstract to be more descriptive

0.103521  2010-12-18 00:01:23 EST5EDT
    - Added html format

    - Prune files with location = root, so they don't sneak back into
      the dist via GatherDir.

0.103520  2010-12-17 22:51:34 EST5EDT
    - Added the ability to sepcify file name and format. Support for
      three formats included: plaintext, markdown, and pod.
    - Initial release after forking from ReadmeFromPod. This module
      remains backward-compatible.


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Automatically convert POD to a README in any format for Dist::Zilla",
  "AUTHOR" => "Ryan C. Thompson <rct\@thompsonclan.org>, Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-ReadmeAnyFromPod",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::ReadmeAnyFromPod",
  "PREREQ_PM" => {
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::FilePruner" => 0,
    "Dist::Zilla::Role::FileWatcher" => 0,
    "Encode" => 0,
    "IO::Handle" => 0,
    "IO::Scalar" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Autobox" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "MooseX::Has::Sugar" => 0,
    "PPI::Document" => 0,
    "Path::Tiny" => "0.004",
    "Pod::Markdown" => 0,
    "Pod::Simple::HTML" => "3.23",
    "Pod::Simple::Text" => "3.23",
    "Scalar::Util" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Carp" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.94",
    "Test::Most" => 0,
    "Test::Requires" => 0,
    "autodie" => 0,
    "lib" => 0,
    "utf8" => 0
  },
  "VERSION" => "0.142470",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::AfterRelease" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::FilePruner" => 0,
  "Dist::Zilla::Role::FileWatcher" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Encode" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IO::Scalar" => 0,
  "IPC::Open3" => 0,
  "List::Util" => 0,
  "Moose" => 0,
  "Moose::Autobox" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "MooseX::Has::Sugar" => 0,
  "PPI::Document" => 0,
  "Path::Tiny" => "0.004",
  "Pod::Markdown" => 0,
  "Pod::Simple::HTML" => "3.23",
  "Pod::Simple::Text" => "3.23",
  "Scalar::Util" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.94",
  "Test::Most" => 0,
  "Test::Requires" => 0,
  "autodie" => 0,
  "lib" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



